lib/chef/provisioning/convergence_strategy/install_sh.rb

```diff
@@ -64,7 +64,18 @@ def setup_convergence(action_handler, machine)
 
         # Install chef client
         # TODO ssh verification of install.sh before running arbtrary code would be nice?
-        @@install_sh_cache[install_sh_url] ||= Net::HTTP.get(URI(install_sh_url))
+        if bootstrap_env.empty?
```

Comment:
@afiune any chance you can do `if convergence_options[:bootstrap_proxy]` instead?  It took me a while to grok what was going on since bootstrap_env isn't used in the if or else :)

Other than that, :+1: